core/amber/src/main/python/core/models/marker.py

```diff
@@ -1,11 +1,70 @@
 from dataclasses import dataclass
+from pyarrow import Table
+from pandas import DataFrame
+from typing import Optional
+from .schema import Schema, AttributeType
+from .schema.attribute_type import FROM_PYOBJECT_MAPPING
 
 
 @dataclass
 class Marker:
     pass
 
 
+@dataclass
+class StartOfUpstream(Marker):
+    pass
+
+
 @dataclass
 class EndOfUpstream(Marker):
     pass
+
+
+@dataclass
+class State(Marker):
+    def __init__(
+        self, table: Optional[Table] = None, pass_to_all_downstream: bool = False
+    ):
+        if table is None:
+            self.data = {}
+            self.schema = Schema()
+            self.add("passToAllDownstream", pass_to_all_downstream)
```

Comment:
how about just use `self.__dict__` to store? so that you don't need to explicitly add `passToAllDownstream` to self.data.

Reply:
Fixed.